src/lib/y2storage/disk_size.rb

```diff
@@ -615,6 +615,32 @@ def to_human_string
       format("%.2f %s", size, unit)
     end
 
+    # A human readable representation that does not exceed the exact size.
+    #
+    # If we have 4.999 GiB of space and prefill the "Size" widget
+    # with a "5.00 GiB" it will then fail validation. We must round down.
+    #
+    # @see to_human_string
+    # @see human_ceil
+    def human_floor
+      return "unlimited" if unlimited?
+      float, unit_s = human_string_components
+      "#{(float * 100).floor / 100.0} #{unit_s}"
+    end
+
+    # A human readable representation that is at least the exact size.
+    #
+    # (This seems unnecessary because actual minimum sizes
+    # have few significant digits, but we use it for symmetry)
+    #
+    # @see to_human_string
+    # @see human_floor
+    def human_ceil
+      return "unlimited" if unlimited?
+      float, unit_s = human_string_components
+      "#{(float * 100).ceil / 100.0} #{unit_s}"
```

Comment:
Code climate does not like this duplication.

I think we could solve this by
```rb
def to_human_string(round_method = :round)
  return "unlimited" if unlimited?
  float, unit_s = human_string_components
  rounded = (float * 100).public_send(round_method) / 100.0
  format("%.2f %s", rounded, unit_s) # the #format is necessary to get "5.00 B" instead of "5.0 B"
end

def human_ceil
  to_human_string(:ceil)
end
```